llama_index/indices/postprocessor/relevancy/base.py

```diff
@@ -0,0 +1,214 @@
+            if self.scorer.agg_mode == AggMode.INTRA_NODE:
+                relevancy_scored = []
+                for nodes in self._subnodes:
+                    relevancy_scored.append(self.scorer.score(nodes))
+                self._subnodes = relevancy_scored
+            elif self.scorer.agg_mode == AggMode.CROSS_NODE:
+                self._subnodes = self.scorer.score_nested(self._subnodes)
+            else:
+                raise ValueError("Invalid scorer agg mode.")
+
+            # Filter the subnodes by considering the population of subnodes
+            for filter in self.filters:
+                if filter.agg_mode == AggMode.INTRA_NODE:
+                    filtered = []
+                    for nodes in self._subnodes:
+                        filtered.append(filter.filter(nodes))
+                    self._subnodes = filtered
+                elif filter.agg_mode == AggMode.CROSS_NODE:
+                    self._subnodes = filter.filter_nested(self._subnodes)
+                else:
+                    raise ValueError("Invalid filter agg mode.")
+
+            # synthesize the subnodes into nodes
+            # Modify the original nodes in place, utilizing
+            # information from the subnodes.
+            self._nodes = self._original_nodes
+            for synthesizer in self.node_synthesizers:
+                if synthesizer.op_mode == MapMode.SUB_NODE:
+                    synthesized = []
+                    for nodes in self._subnodes:
+                        synthesized.append(synthesizer.synthesize(nodes))
+                    self._subnodes = synthesized
+                if self.synthesizer.op_mode == MapMode.NODE:
+                    self._nodes = synthesizer.synthesize(self._nodes)
+                elif synthesizer.op_mode == AggMode.INTRA_NODE:
+                    self._subnodes, self._nodes = synthesizer.synthesize_nested(
+                        self._subnodes, self._nodes
+                    )
+                else:
+                    raise ValueError("Invalid synthesizer operator mode.")
+
+        return self._nodes
+
+    def _aprocess(self):
+        pass
+
+    def _aget_relevancy(self, query, results):
+        pass
+
+
+class OperatorMode(Enum):
+    pass
+
+
+class AggMode(OperatorMode, Enum):
+    """
+    Performs an agg operation on a node level,
+    across all nodes. Splitter must be `None`.
+    """
+
+    INTER_NODE = "inter_node"
+
+    """
+    Performs an agg operation within a node, 
+    on a sub-node level. Splitter must not be `None`.
+    """
+    INTRA_NODE = "intra_node"
+
+    """
+    Performs an agg operation on a sub-node level, 
+    across all nodes. Splitter must not be `None`.
+    """
+    CROSS_NODE = "cross_node"
+
+
+class MapMode(OperatorMode, Enum):
+    """
+    Performs a map operation on the node level.
+    """
+
+    NODE = "node"
+
+    """
+    Performs a map operation on the sub-node level.
+    """
+    SUB_NODE = "sub_node"
+
+
+class Splitter:
```

Comment:
Maybe this can be `llama_index.bridge.langchain.TextSplitter`?

Reply:
Not sure if overlap chunks are a good practice here... It could be OK if you are doing some post formatting.

Reply:
We can just create a simpler wrapper.